### PR TITLE
Typescript support, Add Changelog through changesets, better build pipeline, etc

### PR DESCRIPTION
**Pull request by nicolasmelo1**
- Typescript as first class citzen of the library
- Remove dependency from needing to import `package.json`, it gets added during build time.
- Better build pipeline from the use of changesets. The package can receive contributions from many contributors and each of them will individually appear on the Changelog. I recommend installing [changesets/bot](https://github.com/changesets/bot) to the repo so no PR can be opened without it.
- A simple CONTRIBUTING.md
- Changed a few stuff on your package.json. Now it correctly defines `main`, `module` and `types`. Main is for commonJs, Module for browsers or bundlers and types is for Typescript, duh.
Now the package we deploy to NPM will just contain `dist`, `README`, `LICENSE` and `package.json`, nothing else will exist on the users `node_modules`.

I think thats it.
Hope you like it 😄

### CODE REVIEW

**Comment by meiazero** [comment]
You're using `tsconfig.json` with default options, why not use [tsconfig.guide](https://tsconfig.guide/)?

**Comment by ChristoPy** [comment]
> ###
> **[meiazero](/meiazero) ** commented [Dec 8, 2024](#issuecomment-2526344583)

Trust me, he knows things we don't know yet haha

**Comment by Cristuker** [comment]
Would it be overengineered to separate some functions in other files like requests and do the same for types?

File: package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "define:change": "changeset",
     "release:apply": "changeset version",
     "release:publish": "changeset publish",
+    "lint": "npx @biomejs/biome check --write --unsafe ./src",
```
Comment (meiazero) [inline package.json:18]: It is not necessary to use `npx @biomejs/biome` just `biome`, the script should be `“lint”:“biome check --write --unsafe ./src”,`

Comment (nicolasmelo1) [reply package.json]: Fixed!

**Comment by ChristoPy** [comment]
> @Cristuker I moved requests to a separate file and created a currying function for that. It's a function that returns a function, i think it's a lot better now. Also the types live on types.ts file as it was before
> 
> @meiazero Used tsconfig.guide for the tsconfig.json now
> 
> Other changes:
> 
> * Added Biome.js for linting
> * Created a constants.ts file for storing all the constants
> * Created a proper AbacatePayError for AbacatePay Exceptions
> 
> Probably some changes: AbacatePayError points to the doc, i think the doc link is wrongly pointed to https://www.abacatepay.com/ do you have a better link @ChristoPy?
> 
> I think it seems awesome now.

Link for the docs: [abacatepay.readme.io](https://abacatepay.readme.io)

**Comment by nicolasmelo1** [comment]
@ChristoPy done

File: biome.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "clientKind": "git",
+    "useIgnoreFile": false
+  },
+  "files": {
+    "ignoreUnknown": false,
+    "ignore": ["dist", "README", "LICENSE", "package.json", "tsconfig.json"]
+  },
+  "formatter": {
+    "enabled": true,
+    "indentStyle": "tab"
```
Comment (rafa-thayto) [inline biome.json:14]:
```suggestion
    "indentStyle": "space"
```

Comment (rafa-thayto) [reply biome.json:14]: space > tab, it's simple

Comment (nicolasmelo1) [reply biome.json:14]: +1 for @rafa-thayto. I like tabs, but space is more natural and consume less space on the screens

Comment (fm1randa) [reply biome.json:14]: Got it, thanks

**Pull request merged**